src/lib/retry-logic.ts

```diff
@@ -0,0 +1,71 @@
+import { AxiosError, AxiosRequestConfig } from 'axios';
+import isRetryAllowed from 'is-retry-allowed';
+import { UserPayload } from '../services/user-payload';
+import { ChtApi } from './cht-api';
+
+const RETRY_COUNT = 4;
+
+export const axiosRetryConfig = {
+  retries: RETRY_COUNT,
+  retryDelay: () => 1000,
+  retryCondition: (err: AxiosError) => {
+    const status = err.response?.status;
+    return (!status || status >= 500) && isRetryAllowed(err);
+  },
+  onRetry: (retryCount: number, error: AxiosError, requestConfig: AxiosRequestConfig) => {
+    console.log(`${requestConfig.url} failure. Retrying (${retryCount})`);
+  },
+};
+
+export async function retryOnUpdateConflict<T>(funcWithPut: () => Promise<T>): Promise<T> {
```

Comment:
Is it possible to also handle this in the retry condition defined in the retry config?

Reply:
When an update conflict happens, the `rev` in the doc is outdated.  The correct way to resolve them is to GET, write, and PUT the object again.  If we just retry the PUT, we will either conflict again or overwrite somebody else's changes (in our case, sentinel).  This is why I handle separately.

Reply:
aah i see, we do need to do the GET first.